flink-connector-vitess-cdc/src/main/java/com/ververica/cdc/connectors/vitess/table/VitessTableFactory.java

```diff
@@ -136,13 +145,12 @@ public DynamicTableSource createDynamicTableSource(DynamicTableFactory.Context c
         String tableName = config.get(TABLE_NAME);
         String username = config.get(USERNAME);
         String password = config.get(PASSWORD);
-        VtctldConfig vtctldConfig =
-                new VtctldConfig.Builder()
-                        .hostname(config.get(VTCTL_HOSTNAME))
-                        .port(config.get(VTCTL_PORT))
-                        .username(config.get(VTCTL_USERNAME))
-                        .password(config.get(VTCTL_PASSWORD))
-                        .build();
+        String shard = config.get(SHARD);
```

Comment:
For an option which does not have a default value, we should get its value as following.
```java
String shard =config.getOptional(SHARD).orElse(null);
```
Besides this, we should add `@Nullable` in the constructor and the field to remind the developer.
```java
@Nullable
private final String shard;
public VitessTableSource(
            ResolvedSchema physicalSchema,
            int port,
            String hostname,
            String keyspace,
            String tableName,
            String username,
            String password,
            @Nullable String shard,
            String gtid,
            Boolean stopOnReshard,
            Boolean tombstonesOnDelete,
            SchemaAdjustmentMode schemaNameAdjustmentMode,
            TabletType tabletType,
            String pluginName,
            String name,
            Properties dbzProperties) {
......
}
```